parser-typechecker/src/Unison/Codebase/FileCodebase.hs

```diff
@@ -162,27 +164,30 @@ codebase1
    . MonadUnliftIO m
   => Var v
   => BuiltinAnnotation a
-  => S.Format v -> S.Format a -> CodebasePath -> Codebase m v a
+  => Branch.Cache m -> S.Format v -> S.Format a -> CodebasePath -> m (Codebase m v a)
 codebase1 = codebase1' Sync.syncToDirectory
 
 codebase1'
   :: forall m v a
    . MonadUnliftIO m
   => Var v
   => BuiltinAnnotation a
-  => Common.SyncToDir m v a -> S.Format v -> S.Format a -> CodebasePath -> Codebase m v a
-codebase1' syncToDirectory fmtV@(S.Format getV putV) fmtA@(S.Format getA putA) path =
+  => Common.SyncToDir m v a -> Branch.Cache m -> S.Format v -> S.Format a -> CodebasePath -> m (Codebase m v a)
+codebase1' syncToDirectory branchCache fmtV@(S.Format getV putV) fmtA@(S.Format getA putA) path = do
+  termCache <- Cache.semispaceCache 8192
+  typeOfTermCache <- Cache.semispaceCache 8192
+  declCache <- Cache.semispaceCache 1024
```

Comment:
These are the caches for terms, term types, and decls. Currently hardcoded these cache sizes since the config not readily available here.

Reply:
Might want to initialize these earlier, as the result of a successful read should be the same regardless of which codebase it comes from. (Cache poisoning aside.)

Reply:
We discussed this, and if we want caching for the sync operation, we can just pass these caches as a new argument to this spot:

https://github.com/unisonweb/unison/blob/386bdc754a1e06079114424618fe003a2fb2f7c7/parser-typechecker/src/Unison/Codebase/FileCodebase.hs#L192-L193